tox.ini

```diff
@@ -1,16 +1,47 @@
 [tox]
-envlist = py33,py34,py35,py36,coverage,flake8,isort
+envlist = {py33,py34,py35,py36,coverage}-{cext,nocext},flake8,isort
 
 [testenv]
+setenv =
+    cext: WEBSOCKETS_CEXT_BUILD=1
+    nocext: WEBSOCKETS_CEXT_BUILD=0
 deps =
     py33: asyncio
-commands = python -m unittest
+commands =
+    python setup.py clean --all
+    python -c "import glob, os; [os.remove(f) for files in [glob.glob('./websockets/' + ext) for ext in ['*.so', '*.pyd']] for f in files]"
+    cext: python setup.py build_ext --inplace
+    python -m unittest
 
-[testenv:coverage]
+[testenv:coverage-cext]
+whitelist_externals =
+    env
+setenv =
+    WEBSOCKETS_CEXT_BUILD=1
+    CFLAGS=-O0 -coverage
+    LDFLAGS=-coverage
 commands =
+    python setup.py clean --all
+    python -c "import glob, os; [os.remove(f) for files in [glob.glob('./websockets/' + ext) for ext in ['*.so', '*.pyd']] for f in files]"
+    python setup.py build_ext --inplace
     python -m coverage erase
     python -m coverage run --branch --source=websockets -m unittest
+    env WEBSOCKETS_CEXT_DISABLED=1 python -m coverage run -a --branch --source=websockets -m unittest
     python -m coverage report --fail-under=100
+    gcovr -r . -s
+deps =
+    coverage
+    gcovr
+
+[testenv:coverage-nocext]
+setenv =
+    WEBSOCKETS_CEXT_BUILD=0
+commands =
+    python setup.py clean --all
+    python -c "import glob, os; [os.remove(f) for files in [glob.glob('./websockets/' + ext) for ext in ['*.so', '*.pyd']] for f in files]"
```

Comment:
Minor comment unrelated to the main portion of the patch, but it would be good if these two lines could be DRY-ed up somehow. They appear to be copy-pasted three times verbatim.